docs/data/tree-view/rich-tree-view/editing/EditingCallback.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import { TreeItem2 } from '@mui/x-tree-view/TreeItem2';
+import { MUI_X_PRODUCTS } from './products';
+
+export default function EditingCallback() {
+  const [lastEditedItem, setLastEditedItem] = React.useState<string | null>(null);
+
+  return (
+    <Stack spacing={2} sx={{ width: 400 }}>
+      <Typography>{lastEditedItem || 'No item has been edited yet'}</Typography>
+      <Box sx={{ minHeight: 352, minWidth: 250 }}>
+        <RichTreeView
+          items={MUI_X_PRODUCTS}
+          slots={{ item: TreeItem2 }}
+          isItemEditable={() => true}
+          defaultExpandedItems={['grid', 'pickers']}
+          onItemLabelChange={(itemId, label) => {
+            setLastEditedItem(
+              `The label of item ${itemId} has been edited to ${label}`,
+            );
+          }}
```

Comment:
From a quick read, I thought it was providing the prev and next label, because the sentence displayed is 

```
The label of item grid-premium has been edited to @mui/x-da-premium
```

So I'propse to ad the word "id" to the sentence, and maybe use some emphasis to simplify the extraction of the id and new value from the sentence

```suggestion
  const [lastEditedItem, setLastEditedItem] = React.useState<{ itemId: string; label: string } | null>(null);

  return (
    <Stack spacing={2} sx={{ width: 400 }}>
      {lastEditedItem
         ? <Typography>No item has been edited yet</Typography>
         : <Typography>The label of item with id <em>{itemId}</em> has been edited to <em>{label}</em></Typography>
     }
      <Box sx={{ minHeight: 352, minWidth: 250 }}>
        <RichTreeView
          items={MUI_X_PRODUCTS}
          slots={{ item: TreeItem2 }}
          isItemEditable={() => true}
          defaultExpandedItems={['grid', 'pickers']}
          onItemLabelChange={(itemId, label) => setLastEditedItem({itemId, label})}
```